### PR TITLE
[image-picker][android] Support for granular permissions on Android 13

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Add support for [granular permissions](https://developer.android.com/about/versions/13/behavior-changes-13) on Android that were preventing the camera from launching on devices running Android 13

# How
Added support for requesting these permissions

# Test Plan
Tested on a physical device running android 13.

### CODE REVIEW

File: packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt [hunk 1]
```diff
@@ -171,10 +174,17 @@ class ImagePickerModule : Module() {
...
+        READ_MEDIA_IMAGES.takeIf { !writeOnly },
+        READ_MEDIA_VIDEO.takeIf { !writeOnly }
+      ).toTypedArray()
     } else {
-      arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
+      listOfNotNull(
+        Manifest.permission.WRITE_EXTERNAL_STORAGE,
+        Manifest.permission.READ_EXTERNAL_STORAGE.takeIf { !writeOnly }
+      )
+        .toTypedArray()
```
Comment (lukmccall) [inline packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt:187]:
```suggestion
      listOfNotNull(
        Manifest.permission.WRITE_EXTERNAL_STORAGE,
        Manifest.permission.READ_EXTERNAL_STORAGE.takeIf { !writeOnly }
      ).toTypedArray()
```

**Review by lukmccall — APPROVED** [review]
🔥🔥🔥

**Comment by trickeyd** [comment]
Hey all - I can see that this hasn't been added to a release yet so I tried to patch it using patch-package.  But I get a bunch of kotlin errors, so I guess this diff can't stand alone.

Any idea when a release will be made?
or
Any chance you could point me toward some other diffs I may need for my patch?

The errors are below - thanks in advance.

```
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (4, 36): Unresolved reference: READ_MEDIA_IMAGES
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (5, 36): Unresolved reference: READ_MEDIA_VIDEO
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (177, 54): Unresolved reference: TIRAMISU
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (179, 9): Unresolved reference: READ_MEDIA_IMAGES
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (180, 9): Unresolved reference: READ_MEDIA_VIDEO
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (201, 58): Unresolved reference: TIRAMISU
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (216, 57): Type mismatch: inferred type is Unit but Boolean was expected
e: /Users/jamestrickey/projects/goss-mobile-app/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt: (216, 103): Unresolved reference: TIRAMISU
```

**Comment by barthap** [comment]
I guess you need a more recent `compileSdkVersion`, see #20721

**Comment by trickeyd** [comment]
Hey - thanks for your swift response. 

Yikes - adding a patch for expo to bump `compileSdkVersion`, as well as changing it in my main project was not enough to get this to work. Beyond that, I think patching this way is going to get pretty dangerous...

So as far as you know, the only other course of action is to wait for this to be released and upgrade to RN 71?

**Comment by prasannjeet** [comment]
Is there some way we can somehow patch this before release? Thanks!

**Comment by trickeyd** [comment]
Awesome - could you let us know when, or what version this will be released in pls?

**Comment by barthap** [comment]
SDK 48 with this change is going to be released this week.

**Pull request merged**